src/lib.rs

```diff
@@ -1 +1,521 @@
+use std::{error::Error, marker::PhantomData};
+
+use circuits::{
+    interface::LibraGKRLayeredCircuitTr,
+    layered_circuit::{LayeredCircuit, primitives::Evaluation, utils::compute_num_vars},
+};
+use libra_sumcheck::prove_libra_sumcheck;
+use p3_field::{ExtensionField, Field, PrimeField32};
+use poly::{Fields, MultilinearExtension, mle::MultilinearPoly};
+use sum_check::{SumCheck, interface::SumCheckInterface, primitives::SumCheckProof};
+use transcript::Transcript;
+use utils::{
+    EvalNewAddNMulInput, ProveLibraInput, eval_layer_mle_given_wb_n_wc,
+    eval_new_addi_n_muli_at_rb_bc_n_rc_bc, prepare_phase_one_params,
+    prepare_phase_one_params_with_alpha_beta_rb_rc,
+};
+
+pub mod libra_sumcheck;
 pub mod utils;
+
+pub struct Libra<F: Field, E: ExtensionField<F>> {
+    _marker: PhantomData<(F, E)>,
+}
+
+pub struct LibraProof<F: Field, E: ExtensionField<F>> {
+    pub circuit_output: Vec<F>,
+    pub sumcheck_proofs: Vec<SumCheckProof<F, E>>,
+    pub wbs: Vec<E>,
+    pub wcs: Vec<E>,
+}
+
+impl<F: Field, E: ExtensionField<F>> LibraProof<F, E> {
+    pub fn new(
+        output: Vec<F>,
+        sumcheck_proofs: Vec<SumCheckProof<F, E>>,
+        wbs: Vec<E>,
+        wcs: Vec<E>,
+    ) -> LibraProof<F, E> {
+        LibraProof {
+            circuit_output: output,
+            sumcheck_proofs,
+            wbs,
+            wcs,
+        }
+    }
+}
+
+impl<F: Field + PrimeField32, E: ExtensionField<F>> Libra<F, E> {
+    pub fn prove(circuit: &LayeredCircuit, output: Evaluation<F>) -> LibraProof<F, E> {
+        // Initialize prover transcript
+        let mut transcript = Transcript::init();
+
+        let mut sumcheck_proofs = vec![];
+
+        let mut wbs = vec![];
+
+        let mut wcs = vec![];
+
+        // Get the output vector
+        let output_evals: Vec<Fields<F, E>> = if output.layers[circuit.layers.len()].len() <= 1 {
```

Comment:
might be cleaner to first convert all output elements to base field, then have an if statement at the end that checks if only one output (then pushes a base field zero in that case)